test/functional/feature_masterdatadir.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the master datadir feature.
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    connect_nodes,
+    get_datadir_path,
+    mine_large_block,
+)
+import os
+
+# Copied from feature_pruning.py; should probably be moved into util.py
+def calc_usage(blockdir):
+    return sum(os.path.getsize(blockdir+f) for f in os.listdir(blockdir) if os.path.isfile(os.path.join(blockdir, f))) / (1024. * 1024.)
+
+class MasterDatadirTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+
+    def setup_nodes(self):
+        self.masterdatadir = get_datadir_path(self.options.tmpdir, 0)
+        # Create node 0 to mine.
+        # Create node 1 to test -masterdatadir.
+        self.extra_args = [[],
+                           ["-masterdatadir=" + self.masterdatadir]]
+        self.add_nodes(2, self.extra_args)
+        self.start_node(0)
+
+        self.blocks0 = os.path.join(self.nodes[0].datadir, 'regtest', 'blocks', '')
+        self.blocks1 = os.path.join(self.nodes[1].datadir, 'regtest', 'blocks', '')
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def run_test(self):
+        self.log.info("Mining large blocks until we have more than one blk file")
+        # Make stuff
+        self.nodes[0].generate(150)
+        assert_equal(150, self.nodes[0].getblockcount())
+        # We want more than one block file, so make a few large blocks
+        for i in range(150):
+            mine_large_block(self.nodes[0])
+            if i % 50 == 0:
+                self.log.info("... " + str(i) + " / 150")
```

Comment:
Use placeholders in the log message like this:

```
self.log.info("... %s / 150", i)
```

It is the preferred form since it allows for lazy evaluation.

Applies to all logging calls in this PR :-)

Reply:
Fixed!